spec/controllers/api/v2/app_identifier_visitor_configs_controller_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe Api::V2::AppIdentifierVisitorConfigsController do
+  describe "#show" do
+    let(:app) { FactoryBot.create(:app) }
+    let(:identifier_type) { FactoryBot.create(:identifier_type, name: "clown_id") }
+    let(:feature_gate) { FactoryBot.create(:feature_gate, name: "blab_enabled", registry: { false: 50, true: 50 }) }
+    let(:visitor) { FactoryBot.create(:visitor) }
+    let(:identifier) { FactoryBot.create(:identifier, visitor: visitor) }
+
+    it "has knocked-out weightings and it doesn't include a non-force assignment for a feature gate that isn't feature complete" do
+      FactoryBot.create(
+        :assignment,
+        visitor: visitor,
+        split: feature_gate,
+        variant: "true",
+        context: "the_context",
+        mixpanel_result: "success"
+      )
+
+      get :show, params: {
+        app_name: app.name,
+        version_number: "1.0",
+        build_timestamp: "2019-04-16T14:35:30Z",
+        identifier_type_name: identifier.identifier_type.name,
+        identifier_value: identifier.value
+      }
+
+      expect(response).to have_http_status :ok
+      expect(response_json['splits'][0]['name']).to eq('blab_enabled')
+      expect(response_json['splits'][0]['weights']).to eq('false' => 100, 'true' => 0)
```

Comment:
i just realized that we also wanna make one more change here to convert this into an array of variant objects instead of a map!

Reply:
done!